cli/src/generate/prepare_grammar/expand_tokens.rs

```diff
@@ -167,6 +167,14 @@ impl NfaBuilder {
                 }
                 Ok(true)
             }
+            Rule::Immediate(element) => {
+                self.is_sep = false;
+                if self.expand_rule(element, next_state_id)? {
+                    Ok(true)
+                } else {
+                    Ok(false)
+                }
+            }
```

Comment:
Here we need to somehow instruct the parser to not `SKIP` and immediately `ADVANCE` to the next rule.

Reply:
Maybe we should define a new action like `REJECT_NEXT_EXTRA`, so the lexer rejects if the next lookahead is extra like space.